src/node/internal/internal_fs_utils.ts

```diff
@@ -0,0 +1,572 @@
+  defaultOptions: Record<string, unknown> = {}
+): Record<string, unknown> {
+  if (options == null || typeof options === 'function') {
+    return defaultOptions;
+  }
+
+  if (typeof options === 'string') {
+    defaultOptions = { ...defaultOptions };
+    defaultOptions.encoding = options;
+    options = defaultOptions;
+  } else if (typeof options !== 'object') {
+    throw new ERR_INVALID_ARG_TYPE('options', ['string', 'Object'], options);
+  }
+
+  if (options.encoding !== 'buffer') assertEncoding(options.encoding);
+
+  if (options.signal !== undefined) {
+    validateAbortSignal(options.signal, 'options.signal');
+  }
+
+  return options;
+}
+
+export function copyObject<T>(source: Record<string, T>): Record<string, T> {
+  const target: Record<string, T> = {};
+  for (const key in source) target[key] = source[key] as T;
+  return target;
+}
+
+export function getValidMode(
+  mode: number | undefined,
+  type: 'copyFile' | 'access'
+): number {
+  let min = kMinimumAccessMode;
+  let max = kMaximumAccessMode;
+  let def = F_OK;
+  if (type === 'copyFile') {
+    min = kMinimumCopyMode;
+    max = kMaximumCopyMode;
+    def = mode || kDefaultCopyMode;
+  } else {
+    strictEqual(type, 'access');
+  }
+  if (mode == null) {
+    return def;
+  }
+  validateInteger(mode, 'mode', min, max);
+  return mode;
+}
+
+const defaultCpOptions: CopyOptions = {
+  dereference: false,
+  errorOnExist: false,
+  force: true,
+  preserveTimestamps: false,
+  recursive: false,
+  verbatimSymlinks: false,
+};
+
+export function validateCpOptions(
+  _options: unknown
+): CopyOptions | CopySyncOptions {
+  if (_options === undefined) return { ...defaultCpOptions };
+  validateObject(_options, 'options');
+  const options: CopyOptions = { ...defaultCpOptions, ..._options };
+  validateBoolean(options.dereference, 'options.dereference');
+  validateBoolean(options.errorOnExist, 'options.errorOnExist');
+  validateBoolean(options.force, 'options.force');
+  validateBoolean(options.preserveTimestamps, 'options.preserveTimestamps');
+  validateBoolean(options.recursive, 'options.recursive');
+  validateBoolean(options.verbatimSymlinks, 'options.verbatimSymlinks');
+  options.mode = getValidMode(options.mode, 'copyFile');
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-boolean-literal-compare
+  if (options.dereference === true && options.verbatimSymlinks === true) {
+    throw new ERR_INCOMPATIBLE_OPTION_PAIR('dereference', 'verbatimSymlinks');
+  }
+  if (options.filter !== undefined) {
+    // eslint-disable-next-line @typescript-eslint/unbound-method
+    validateFunction(options.filter, 'options.filter');
+  }
+  return options;
+}
+
+// converts Date or number to a fractional UNIX timestamp
+export function toUnixTimestamp(
+  time: string | number | Date,
+  name: string = 'time'
+): string | number | Date {
+  // @ts-expect-error TS2367 number to string comparison error.
+  if (typeof time === 'string' && +time == time) {
```

Comment:
It might be nice to be more explicit here, as it's difficult to follow the logic. `Number(time).toString() === time` perhaps?

Reply:
This method isn't likely to be retained actually.